crates/boundless-market/src/contracts/boundless_market.rs

```diff
@@ -1381,6 +1392,132 @@ pub fn decode_calldata(data: &Bytes) -> Result<(Vec<Fulfillment>, AssessorReceip
     ))
 }
 
+#[derive(Debug, Clone)]
+/// Represents the parameters for submitting a Merkle Root.
+pub struct Root {
+    /// The address of the set verifier contract.
+    pub verifier_address: Address,
+    /// The Merkle root of the proof.
+    pub root: B256,
+    /// The seal of the proof.
+    pub seal: Bytes,
+}
+
+#[derive(Debug, Clone)]
+/// Represents the parameters for pricing a request.
+pub struct Price {
+    /// The list of requests to be priced.
+    pub requests: Vec<ProofRequest>,
+    /// The list of client signatures for the requests.
+    pub client_sigs: Vec<Bytes>,
+}
+
+#[derive(Clone)]
+/// Represents the parameters for a fulfillment.
+pub struct Fulfill<P> {
+    /// The Boundless market service instance.
+    pub boundless_market: BoundlessMarketService<P>,
+    /// The parameters for submitting a Merkle Root
+    pub root: Option<Root>,
+    /// The parameters for pricing a request
+    pub price: Option<Price>,
+    /// The fulfillments to be submitted
+    pub fulfillments: Vec<Fulfillment>,
+    /// The assessor receipt
+    pub assessor_receipt: AssessorReceipt,
+    /// Whether to withdraw the fee
+    pub withdraw: bool,
+}
+
+impl<P: Provider> Fulfill<P> {
```

Comment:
I like this, but I think its important we have consistency across the SDK with the work @nategraf is doing, so definitely sync with him!

Also I think something like: FulfillmentClient, FulfillmentHelper, FulfillmentBuilder would be better